app/assets/javascripts/ama_layout/forms/placeholder.coffee

```diff
@@ -0,0 +1,30 @@
+class AMALayout.Placeholder
+  constructor: (args = {}) ->
+    args = $.extend({}, @defaultVariables(), args)
+    @placeholderTextColor = args.placeholderTextColor
+    @inputTextColor = args.inputTextColor
+    @defaultTextList = args.defaultTextList
+    @setup()
+
+  setup: ->
+    for defaultText in @defaultTextList
+      @changeDefaultPlaceholderColor(defaultText)
+      @updatePlaceholderColor(defaultText)
+
+  changeDefaultPlaceholderColor: (defaultText) ->
+    $("option:selected:contains('#{defaultText}')").parent('select').css('color', @placeholderTextColor)
```

Comment:
Have a look at this: https://github.com/amaabca/ama_layout/compare/placeholder-js...refactoring-placeholder

It reuses the "index == 0" logic, instead of looking for the selected option.